code/modules/power/lighting.dm

```diff
@@ -312,12 +314,15 @@
 			if(rigged)
 				if(status == LIGHT_OK && trigger)
 					explode()
-			else if( prob( min(60, switchcount*switchcount*0.01) ) )
+			else if( prob( min(60, (switchcount^2)*0.01) ) )
 				if(trigger)
 					burn_out()
 			else
 				use_power = ACTIVE_POWER_USE
-				set_light(brightness, bulb_power, bulb_colour)
+				if(!nightshift_enabled)
+					set_light(brightness, bulb_power, bulb_colour)
+				if(nightshift_enabled)
+					set_light(brightness, SSnightshift.nightshift_light_power, SSnightshift.nightshift_light_color)
```

Comment:
Pulling colors directly from SSnightshift was a temporary hack to get around lights always being forced to #FFFFFF. You should probably add nightshift_bulb_power and nightshift_bulb_colour vars to lights and remove the nightshift_light_power and nightshift_light_color vars from SSnightshift.